ansible/library/announce_routes.py

```diff
@@ -330,44 +472,123 @@ def fib_t1_lag(topo, ptf_ip, no_default_route=False, action="announce"):
             change_routes(action, ptf_ip, port, routes_vips)
 
 
-def fib_m0(topo, ptf_ip, no_default_route=False, action="announce"):
+def get_new_ip(curr_ip, skip_count):
+    """
+    Get the [skip_count]th ip after curr_ip
+    """
+    new_ip = ipaddress.ip_address(int(curr_ip) + skip_count)
+    return new_ip
+
+
+def get_next_ip_by_net(net_str):
+    """
+    Get the nearest next non-overlapping ip address based on the net_str
+    Sample input:
+    "192.168.0.1/24"
+    """
+    net = ipaddress.ip_network(UNICODE_TYPE(net_str), strict=False)
+    net_size = int(net.broadcast_address) + 1 - int(net.network_address)
+    next_net = get_new_ip(net.network_address, net_size)
+    return next_net
+
+
+def get_next_net_of_nets(nets):
+    """
+    Get the nearest next non-overlapping ip address based on the network list
+    Sample input:
+    [
+        "192.168.0.1/24",
+        "192.168.0.1/25",
+        "192.168.0.128/25"
+    ]
+    """
+    max_next_ip = None
+    for vlan in nets:
+        next_ip = get_next_ip_by_net(vlan)
+        if max_next_ip is None:
+            max_next_ip = next_ip
+        elif next_ip > max_next_ip:
+            max_next_ip = next_ip
```

Comment:
What is the expected output if input looks like below? `192.168.1.1/24` or `192.168.3.1/24`?
```
    [
        "192.168.0.1/24",
        "192.168.2.1/24",
    ]
```